### DISCUSSION TITLE
Motion log file gets deleted automatically after reboot

### DISCUSSION THREAD

**Discussion by tqchiem** [Motion-Project/motion#1800] [General]
I installed Motion 4.3.2 on my Raspberry Pi 4B/Ubuntu 22.04LTS server 64-bit. I created /var/log/motion.log with "motion:motion" ownership. Motion works and also logs events in the log without any problem. The log is clean, no issue at all. However, after I reboot the system, Motion ran into some errors, as below:

```
motion[1578]: [0:motion] [ERR] [ALL] myfopen: Error opening file /var/log/motion.log with mode a: Permission denied
motion[1578]: [0:motion] [EMG] [ALL] motion_startup: Exit motion, cannot create log file /var/log/motion.log: Permission denied
```

and /var/log/motion.log disappeared.

In my /etc/sudoers, I have "motion ALL=(ALL) NOPASSWD:ALL"

Why does it delete the log and then complain about file opening permission and try to create the file?.

Any help would be very much appreciated. Thank you.

**Comment by tosiara** [comment]
If you put the log into `/var/log` you need to ensure that motion user has write access to `/var/log`.
You can test like this:
```
$ sudo touch /var/log/motion.log
$ sudo chown motion:motion /var/log/motion.log
$ echo 123 > /var/log/motion.log
$ rm /var/log/motion.log
rm: cannot remove '/var/log/motion.log': Permission denied
```

**Reply by tqchiem** [reply]
My motion.log file was created that way and by default it has 644 permissions (rw/r/r). I also tried chown motion:adm to match with motion default install settings. 

Motion default installation creates motion directory with motion.log in it, having ownership as motion:adm. I replicated that and restart motion. It logs messages without any problem. However, after reboot, it just deleted the whole directory and systemctl status motion shows the following:

```
Aug 12 12:13:17 DevUnit motion[1553]: [0:motion] [NTC] [ALL] config_camera: Processing camera config file /usr/cvconf/camera3.conf
Aug 12 12:13:17 DevUnit motion[1553]: [0:motion] [ERR] [ALL] create_path: Problem creating directory /var/log/motion: Permission denied
Aug 12 12:13:17 DevUnit motion[1553]: [0:motion] [EMG] [ALL] motion_startup: Exit motion, cannot create log file /var/log/motion/motion.log: Permission denied
Aug 12 12:13:17 DevUnit motion[1553]: [0:motion] [NTC] [ALL] config_camera: Processing camera config file /usr/cvconf/camera0.conf
Aug 12 12:13:17 DevUnit motion[1553]: [0:motion] [NTC] [ALL] config_camera: Processing camera config file /usr/cvconf/camera1.conf
Aug 12 12:13:17 DevUnit motion[1553]: [0:motion] [NTC] [ALL] config_camera: Processing camera config file /usr/cvconf/camera2.conf
Aug 12 12:13:17 DevUnit motion[1553]: [0:motion] [NTC] [ALL] config_camera: Processing camera config file /usr/cvconf/camera3.conf
Aug 12 12:13:17 DevUnit motion[1553]: [0:motion] [ERR] [ALL] create_path: Problem creating directory /var/log/motion: Permission denied
Aug 12 12:13:17 DevUnit motion[1553]: [0:motion] [EMG] [ALL] motion_startup: Exit motion, cannot create log file /var/log/motion/motion.log: Permission denied
Aug 12 12:13:17 DevUnit systemd[1]: motion.service: Deactivated successfully.
```
Not sure why it wants to create the motion directory when it was created before reboot. Who would delete motion?. The /var/log structure is on SD card.

As I understand it, the motion.log file can be anywhere as long as it is referenced in the motion.conf configuration file correctly; i.e., `log_file /var/log/motion/motion.log`.

I noticed the default motion directory has the following:
`drwxr-s---   2 motion    adm                4096 Jun 26 21:24 motion`

It's a very strange problem.

**Comment by olddog55** [comment]
I am guessing here, but it looks like you may be caught up in systemd's crazy concept around logging, /var/log and the file-cleanup services.  You might try avoiding /var/log and doing the motion logging directly into the motion application's directory.  And adjust any motion log rotation configurations accordingly.

**Reply by tqchiem** [reply]
I think your guess is a very good guess.
 
Since then I run a script to create "/var/motion.log" with motion ownership just prior to starting motion. It is a patch but it works every time after reboot.

My system uses log2ram to use RAM for logging, instead of SD card. I suspect log2ram might have something to do with it, but I cannot prove that. Log2ram keeps a copy of /var/log on the SD card (at "/var/hdd.log") which has motion.log in it. During troubleshooting, I ran this log2ram before reboot to make sure it has motion.log. After reboot, it disappears. As I understand log2ram does load /var/hdd.log to RAM after reboot. 

Maybe the system startup order is causing this problem; i.e. motion first followed by log2ram. My patch works but I lose the pre-reboot logs. When I have time, I will investigate this further.
